js/subprocess.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as dispatch from "./dispatch";
+import * as flatbuffers from "./flatbuffers";
+import * as msg from "gen/msg_generated";
+import { assert, unreachable } from "./util";
+import { close, File } from "./files";
+import { ReadCloser, WriteCloser } from "./io";
+
+/** How to handle subsubprocess stdio.
+ *
+ * "inherit" The default if unspecified. The child inherits from the
+ * corresponding parent descriptor.
+ *
+ * "piped"  A new pipe should be arranged to connect the parent and child
+ * subprocesses.
+ *
+ * "null" This stream will be ignored. This is the equivalent of attaching the
+ * stream to /dev/null.
+ */
+export type SubprocessStdio = "inherit" | "piped" | "null";
```

Comment:
Not a fan of the fact that 'piped' is either a writable or readable end of a pipe, depending on the property it is used for.

Reply:
Ya - it's much less flexible than the general case... In particular we also cannot supply arbitrary FDs. However, this is what's easy to do now because tokio_process exposes this interface. 
Note that `Subprocess.stdin` is an `io.WriteCloser` and `Subprocess.stdout` is `io.ReadCloser` - so it is still type safe.